src/www/ui/api/Helper/DbHelper.php

```diff
@@ -356,6 +356,78 @@ public function getJobs($id = null, $limit = 0, $page = 1, $uploadId = null)
     return [$jobs, $totalResult];
   }
 
+  /**
+   * @brief Get the recent jobs created by an user.
+   *
+   * If a limit is passed, the results are trimmed. If an ID is passed, the
+   * information for the given id is only retrieved.
+   *
+   * @param integer $id       Set to get information of only given job id
+   * @param integer $uid      Set to get information of only given user's ID
+   * @param integer $limit    Set to limit the result length
+   * @param integer $page     Page number required
+   * @param integer $uploadId Upload ID to be filtered
+   * @return array[] List of jobs at first index and total number of pages at
+   *         second.
+   */
+  public function getUserJobs($id = null, $uid=null, $limit = 0, $page = 1, $uploadId = null)
+  {
+    $jobSQL = "SELECT job_pk, job_queued, job_name, job_upload_fk," .
+      " job_user_fk, job_group_fk FROM job WHERE job_user_fk='$uid'";
+    $totalJobSql = "SELECT count(*) AS cnt FROM job WHERE job_user_fk='$uid'";
+
+    $filter = "";
+    $pagination = "";
+
+    $params = [];
```

Comment:
The user id should be passed as an parameter to the PostgreSQL server and not embedded in the query to prevent query injection.

```suggestion
    $jobSQL = "SELECT job_pk, job_queued, job_name, job_upload_fk," .
      " job_user_fk, job_group_fk FROM job WHERE job_user_fk=$1";
    $totalJobSql = "SELECT count(*) AS cnt FROM job WHERE job_user_fk=$1";

    $filter = "";
    $pagination = "";

    $params = [];
    $params[] = $uid;
```

Reply:
Resloved